src/utils/parseIntegerFromFormattedString.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * This takes in an English or German formatted (float or integer looking) amount as a string and converts it to an integer (cent amount)
+ * @returns amount in cents
+ */
+export function parseIntegerFromFormattedString( amountStr: string ): number {
+	// Replace all commas with dots and then remove all dots except for the last one
+	amountStr = amountStr.replace( /,/g, '.' );
+	amountStr = amountStr.replace( /[.](?=.*[.])/g, '' );
+
+	// Remove all non-numeric chars
+	amountStr = amountStr.replace( /[^0-9.]/g, '' );
+
+	const amountNumber = Number( amountStr );
+	if ( isNaN( amountNumber ) ) {
+		return 0;
+	}
+
+	// turn euro float into integer (cent) amount
+	return Math.trunc( amountNumber * 100 );
```

Comment:
Just FYI, I was about to suggest a different, mores string-based approach, also described in the ticket: splitting at the last dot, making sure the 2nd string always has 2 digits  concatenating the strings, and then convertig to number. In PHP that would have given us a proper integer.
However, TIL that [JavaScript stores *all* numbers as 64bit floats](https://medium.com/@vincentcorbee/lets-talk-numbers-in-javascript-72ffe1eb1f3f), so keeping this function almost identical to the one it replaces and just multiplying with 100 is much more sane and less error prone.